### PR TITLE
[Refactor](load) refactor memtable flush actively

### PR DESCRIPTION
**Pull request by sollhui**
## Proposed changes

Refactor memtable flush actively and make it usable and optimize readability.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run p0

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run p0

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

File: be/src/common/config.h [hunk 1]
```diff
@@ -585,8 +585,12 @@ DECLARE_mInt32(memory_maintenance_sleep_time_ms);
 DECLARE_mInt32(memory_gc_sleep_time_ms);
 
 // Sleep time in milliseconds between load channel memory refresh iterations
+// Deprecated
 DECLARE_mInt64(load_channel_memory_refresh_sleep_time_ms);
```
Comment (kaijchen) [inline be/src/common/config.h:589]: Is backward compatibility preserved?

Comment (HHoflittlefish777) [reply be/src/common/config.h:589]: yes

**Comment by HHoflittlefish777** [comment]
run buildall

File: be/src/common/daemon.cpp [hunk 2]
```diff
@@ -276,14 +276,14 @@ void Daemon::memory_gc_thread() {
     }
 }
 
-void Daemon::load_channel_tracker_refresh_thread() {
+void Daemon::memtable_flush_mgr_tracker_refresh_thread() {
     // Refresh the memory statistics of the load channel tracker more frequently,
     // which helps to accurately control the memory of LoadChannelMgr.
-    while (!_stop_background_threads_latch.wait_for(
-                   std::chrono::milliseconds(config::load_channel_memory_refresh_sleep_time_ms)) &&
```
Comment (kaijchen) [inline be/src/common/daemon.cpp:283]: Backward compatibility is broken by just deleting this config.
`load_channel_memory_refresh_sleep_time_ms` should act as an alias of `memtable_flush_mgr_memory_refresh_sleep_time_ms`.

File: be/src/common/config.h [hunk 3]
```diff
@@ -591,8 +591,12 @@ DECLARE_mInt32(memory_maintenance_sleep_time_ms);
...
 
+// Sleep time in milliseconds between memtbale flush mgr memory refresh iterations
+DECLARE_mInt64(memtable_flush_mgr_memory_refresh_sleep_time_ms);
```
Comment (kaijchen) [inline be/src/common/config.h:598]: Maybe call it `memtable_memory_check_interval_ms`?

File: be/src/common/daemon.cpp (hunk 2 above)
Comment (freemandealer) [reply be/src/common/daemon.cpp:283]: Agree. But how to achive 'alias' semantic?

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 53.48 seconds
 stream load tsv:          508 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          81 seconds loaded 1101869774 Bytes, about 12 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162458628 Bytes

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Review by freemandealer — APPROVED** [review]
LGTM, nice job!

File: be/src/olap/delta_writer.h [hunk 4]
```diff
@@ -120,6 +120,8 @@ class DeltaWriter {
 
     int64_t tablet_id() { return _tablet->tablet_id(); }
 
+    int64_t txn_id() { return _req.txn_id; }
```
Comment (kaijchen) [inline be/src/olap/delta_writer.h:123]: Maybe we should not add this interface just for printing logs

File: docs/en/docs/admin-manual/config/be-config.md [hunk 5]
```diff
@@ -958,6 +958,11 @@ BaseCompaction:546859:
 * Description: Whether to use mmap to allocate memory
 * Default value: false
 
+#### `memtable_mem_tracker_refresh_interval_ms`
+
+* Description: Sleep time in milliseconds between memtbale flush mgr refresh iterations
```
Comment (kaijchen) [inline docs/en/docs/admin-manual/config/be-config.md:963]:
```suggestion
* Description: Interval in milliseconds between memtbale flush mgr refresh iterations
```

File: docs/zh-CN/docs/admin-manual/config/be-config.md [hunk 6]
```diff
@@ -971,6 +971,11 @@ BaseCompaction:546859:
 * 描述：是否使用mmap分配内存
 * 默认值：false
 
+#### `memtable_mem_tracker_refresh_interval_ms`
+
+* 描述：memtable主动下刷时刷新内存统计的睡眠时间
```
Comment (kaijchen) [inline docs/zh-CN/docs/admin-manual/config/be-config.md:976]:
```suggestion
* 描述：memtable主动下刷时刷新内存统计的周期（毫秒）
```

File: be/src/olap/delta_writer.h (hunk 4 above)
Comment (HHoflittlefish777) [reply be/src/olap/delta_writer.h:123]: I think it is necessary, so do int64_t tablet_id() { return _tablet->tablet_id(); } above.

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 51.4 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17161989046 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.53 seconds
 stream load tsv:          505 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17162109601 Bytes

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.7 seconds
 stream load tsv:          505 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.8 seconds inserted 10000000 Rows, about 347K ops/s
 storage size: 17162038788 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 52.08 seconds
 stream load tsv:          509 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162330232 Bytes

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.27 seconds
 stream load tsv:          509 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17162379598 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.01 seconds
 stream load tsv:          511 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162339973 Bytes

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.26 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17162349072 Bytes

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.23 seconds
 stream load tsv:          511 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162311880 Bytes

**Comment by HHoflittlefish777** [comment]
run arm

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.21 seconds
 stream load tsv:          510 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162441509 Bytes

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.76 seconds
 stream load tsv:          543 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162172928 Bytes

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.74 seconds
 stream load tsv:          508 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17162446334 Bytes

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.63 seconds
 stream load tsv:          508 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162489463 Bytes

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.76 seconds
 stream load tsv:          504 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162474882 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.42 seconds
 stream load tsv:          502 seconds loaded 74807831229 Bytes, about 142 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162670322 Bytes

**Review by kaijchen — APPROVED** [review]
LGTM

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.14 seconds
 stream load tsv:          510 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17162375740 Bytes

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**